arborator/main.py

```diff
@@ -134,7 +229,8 @@ def write_outliers(outliers,outfile):
 def stage_data(groups,outdir,metadata_df,id_col,max_missing_frac=1):
     files = {}
     for group_id in groups:
-        d = os.path.join(outdir,f"{group_id}")
+        group_id_no_spaces = str(group_id).replace(" ", "_")
```

Comment:
Removal of spaces here is done so that a folder can be created with the group ID correct?

I wonder if we should instead only allow a limited set of characters (e.g., `[A-Za-z0-9_-.]`) and replace all other characters with underscores? That way we avoid any potential issues with weird characters in directory names.

Though issue is that I think some possible partition values may have e.g., `:` or `;` or other special characters (salmonella serotypes). It's also possible that people may have e.g., French named metadata they want to group by, so we'd need to think about supporting that.

Maybe we could have a parameter `--replace_group_special_characters` that defaults to true, but if people needed special characters it can be disabled. Thoughts?